src/userSurvey.js

```diff
@@ -1,118 +1,154 @@
 /**
- * Present surveys to the user.
+ * A module to facilitate surveys of study participants.
+ * 
+ * # User Experience
+ *   * If the user has not been previously prompted for the survey,
+ *     the survey will open in a new tab.
+ *   * The study's browser action popup will contain either a page
+ *     prompting the user complete the survey (with options to open
+ *     the survey or decline the survey), or a neutral page (if the
+ *     user has already completed or declined the survey).
+ *   * If the user has been previously prompted for the survey, and
+ *     has not completed or declined the survey, the user will be
+ *     reminded to complete the survey with a browser notification
+ *     at a set interval.
+ * 
+ * # Limitations
+ * Note that this module is currently very limited: it only supports
+ * one survey per study, with few options and a constrained design.
+ * We have not yet decided whether to build out this module or implement
+ * survey functionality in the Rally core add-on.
+ * 
+ * # Content Security Policy Requirements
+ * This module depends on inline scripts in browser action popups, which
+ * require special Content Security Policy permissions in the extension
+ * manifest (the `"content_security_policy"` key). Those permissions
+ * are currently the following additional `script-src` values.
+ *   * `'sha256-7MkXA5Z7wxRyznhUZN3nVVs9GEQpvRXdYihZZqR2y6w='`
+ *   * `'sha256-l+kgCjP15GJlVSDL9qMNffrHu8mxJcag42o2TYofOUM='`
  * @module webScience.userSurvey
  */
-import * as storage from "./storage.js"
-import * as messaging from "./messaging.js"
+import * as id from "./id.js";
+import * as storage from "./storage.js";
+import * as messaging from "./messaging.js";
+import * as inline from "./inline.js";
+import popupPromptPage from "./html/userSurvey.popupPrompt.html";
```

Comment:
Huh, is it the rollup plugin using this? I've never seen HTML imported like this.

Reply:
Yeah... I came up with this approach to inlining HTML. It's essentially the same as the approach to inlining content scripts.